src/main/java/net/sf/jabref/JabRef.java

```diff
@@ -350,7 +350,7 @@ public void start(String[] args) {
                     }
                     BibDatabaseContext databaseContext = pr.getDatabaseContext();
                     databaseContext.setDatabaseFile(theFile);
-                    Globals.prefs.fileDirForDatabase = databaseContext.getFileDirectory(Globals.FILE_FIELD).toArray(new String[0]);
+                    Globals.prefs.fileDirForDatabase = databaseContext.getFileDirectory().toArray(new String[0]);
```

Comment:
I don't like the toArray stuff. I suggest you change the Array in The GlobalPrefs to a type of List<String>, as Array also implements the List Interface. Then you can omit the toArray stuff. 
And for that 
`                            dirs = new String[] {Globals.prefs.get(Globals.FILE_FIELD + Globals.DIR_SUFFIX)};
`
simply use Arrays.asList